agent/fw_support.c

```diff
@@ -76,8 +76,11 @@ void nr_fw_support_add_package_supportability_metric(
     }
   }
 
-  if ('\0' == major_version[0]) {
+  if ('\0' == major_version[0] && '\0' == package_version[0]) {
```

Comment:
I'm unclear on the logic here - it seems that if the supplied `package_version` is an empty string (or NULL) then we could short circuit out on line 60.